package.json

```
@@ -4,8 +4,7 @@
   "description": "TextMate helpers",
   "main": "./lib/first-mate.js",
   "scripts": {
-    "test": "grunt test",
-    "prepare": "grunt prepublish",
+    "test": "node node_modules/jasmine-focused/bin/jasmine-focused --coffee --captureExceptions spec",
```

I think any module that is an explicitly written dependency in `package.json` can be run by name from the `package.json` scripts.

So I think this can just be:
```suggestion
    "test": "jasmine-focused --coffee --captureExceptions spec",
```

And we have to remember to make sure to compile the `.pegjs` files again, I assume, since it looks like Grunt had been handling that before during the `grunt prepublish` steps. See `Gruntfile.coffee` for the details of that.

---

Great catch there. Yeah they should be able to run like you suggested. And might as well do some cleanup, and add grunt (or another method) to compile the `pegjs` files so we can continue using normal scripts there.

But would you be on board with compiling the `pegjs` via GitHub Action's, rather than a prepublish script. Since I've love to be able to cut down on what we have to do when building the editor itself

---

Well the version of `first-mate` package on the npm registry has the PEGjs compiled already. `lib/scope-selector-parser.js`.

Whatever we need to do to have it as plain JS is pretty much okay by me.

Assuming we do want to use this forked version straight off of GitHub, then it would be fine to have it done as a GitHub action... then pushed to some `dist` branch???

I just thought of something... Any URL pointing to a tarball will do. So if we want to package our tarballs using `npm pack` in GitHub actions and post that packed tarball as a GitHub Releases artifact... we can then use the direct download URL to that tarball in Pulsar's `package.json` and it should work fine. Just a thought.